jdaviz/configs/default/plugins/data_quality/data_quality.py

```diff
@@ -0,0 +1,271 @@
+            if not flag['show'] or
+
+            # hide the flag if `flags_filter` has entries but not this one:
+            (
+                len(self.flags_filter) and
+                not np.isin(list(flag['decomposed'].keys()), list(self.flags_filter)).any()
+            )
+        ])
+
+        with delay_callback(dq_layer.state, 'v_min', 'v_max', 'alpha', 'stretch', 'cmap'):
+            # set correct stretch and limits:
+            # dq_layer.state.stretch = 'lookup'
+            stretch_object = dq_layer.state.stretch_object
+            stretch_object.flags = flag_bits
+            stretch_object.dq_array = dq_layer.get_image_data()
+            stretch_object.hidden_flags = hidden_flags
+
+            # update the colors of the listed colormap without
+            # reassigning the layer.state.cmap object
+            cmap = dq_layer.state.cmap
+            cmap.colors = rgb_colors
+            cmap._init()
+
+            # trigger updates to cmap in viewer:
+            dq_layer.update()
+
+            if len(flag_bits):
+                dq_layer.state.v_min = min(flag_bits)
+                dq_layer.state.v_max = max(flag_bits)
+
+            dq_layer.state.alpha = self.dq_layer_opacity
+
+    def update_visibility(self, index):
+        self.decoded_flags[index]['show'] = not self.decoded_flags[index]['show']
+        self.send_state('decoded_flags')
+        self.update_cmap()
+
+    def vue_update_visibility(self, index):
+        self.update_visibility(index)
+
+    def update_color(self, index, color):
+        self.decoded_flags[index]['color'] = color
+        self.send_state('decoded_flags')
+        self.update_cmap()
+
+    def vue_update_color(self, args):
+        index, color = args
+        self.update_color(index, color)
+
+    @observe('science_layer_selected')
+    def mission_or_instrument_from_meta(self, event):
+        if not hasattr(self, 'science_layer'):
+            return
+
+        layer = self.science_layer.selected_obj
+        if not len(layer):
+            return
+
+        # this is defined for JWST and ROMAN, should be upper case:
+        telescope = layer[0].layer.meta.get('telescope', None)
+
+        if telescope is not None:
+            self.flag_map_selected = telescope_names[telescope.lower()]
+
+    def vue_hide_all_flags(self, event):
+        for flag in self.decoded_flags:
+            flag['show'] = False
+
+        self.send_state('decoded_flags')
+        self.update_cmap()
+
+    def vue_clear_flags_filter(self, event):
+        self.flags_filter = []
+
+    def vue_show_all_flags(self, event):
+        for flag in self.decoded_flags:
+            flag['show'] = True
+
+        self.send_state('decoded_flags')
+        self.flags_filter = []
+
+    @property
+    def user_api(self):
+        return PluginUserApi(
+            self,
+            expose=(
+                'science_layer', 'dq_layer',
+                'decoded_flags', 'flags_filter'
+            )
+        )
```

Comment:
for now, I think we also at least want `viewer` (and possibly `flag_map` - but that would require using a PluginSelectComponent for that instead of handling the items manually here).  How about `dq_layer_opacity`?

I do think it makes sense to hold back on the color/visibility methods if we're not sure of their API yet.... but eventually I'm guessing we'll want those exposed as well.

Reply:
Done in c2cd379.